src/main/java/io/gingersnapproject/service/MembershipListener.java

```diff
@@ -0,0 +1,131 @@
+package io.gingersnapproject.service;
+
+import java.net.InetAddress;
+import java.net.URI;
+import java.net.UnknownHostException;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.CompletionStage;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.event.Observes;
+import javax.enterprise.inject.Instance;
+import javax.inject.Inject;
+
+import io.gingersnapproject.cdc.ReschedulingTask;
+import io.gingersnapproject.cdc.configuration.Configuration;
+
+import io.quarkus.runtime.ShutdownEvent;
+import io.quarkus.runtime.StartupEvent;
+import org.eclipse.microprofile.config.inject.ConfigProperty;
+import org.infinispan.commons.util.concurrent.CompletableFutures;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@ApplicationScoped
+public class MembershipListener {
+
+   private static final Logger log = LoggerFactory.getLogger(MembershipListener.class);
+   private static final ScheduledExecutorService executor = Executors.newSingleThreadScheduledExecutor(r -> new Thread(r,"membership-listener"));
+
+   // Accessed by a single thread.
+   private final Set<InetAddress> members = new HashSet<>();
+   @Inject Configuration configuration;
+
+   // TODO: remove instance wrapper once implementation available.
+   @Inject Instance<DynamicMembershipManagement> dmm;
+
+   @ConfigProperty(name = "gingersnap.dynamic-membership", defaultValue = "false")
+   boolean dynamicMembership;
+
+   private ReschedulingTask<Void> task;
+   private volatile boolean shutdown;
+
+   void onStartup(@Observes StartupEvent ignore) {
+      if (task != null || dmm.isUnsatisfied()) return;
+
+      if (dynamicMembership) {
+         task = new ReschedulingTask<>(executor, this::pollDNSInformation, __ -> false, 15, TimeUnit.SECONDS, __ -> true);
```

Comment:
We should probably make this more frequent. Maybe Poll every second?

Reply:
I think it doesn't change because the DNS records usually are cached. Running with minikube, IIRC takes around 20~30 secs to identify changes.

Reply:
I worry about loss of errors here as we don't log if a DNS error is thrown. Maybe we should log it as DEBUG at least?